packages/x-tree-view/src/internals/plugins/useTreeViewItems/useTreeViewItems.tsx

```diff
@@ -147,6 +143,27 @@ export const useTreeViewItems: TreeViewPlugin<UseTreeViewItemsSignature> = ({
     [instance],
   );
 
+  const isItemEditable = React.useCallback(
+    (itemId: string | null): itemId is string => {
```

Comment:
I think you can remove the `| null` and the first check here
I guess you did the same as on `isItemDisabled`, but now that we correctly type the whole codebase, it's a check that I will remove :ok_hand: